packages/curve-ui-kit/src/features/connect-wallet/lib/wagmi/wagmi-config.ts

```diff
@@ -17,84 +10,45 @@ declare module 'wagmi' {
-        ([id, config]): Chain =>
-          // use the backend data to configure new chains, but use wagmi contract addresses and useful properties/RPCs
-          defineChain({
-            ...(wagmi[id] ?? {
-              nativeCurrency: { name: config.symbol, symbol: config.symbol, decimals: DECIMALS[id] ?? 18 },
-            }),
-            id,
-            testnet: config.isTestnet,
-            name: config.name,
-            rpcUrls: { default: { http: getRpcUrls(id) } },
-            ...(config.explorerUrl && {
-              blockExplorers: { default: { name: new URL(config.explorerUrl).host, url: config.explorerUrl } },
-            }),
-          }),
-      ) as [Chain, ...Chain[]],
-      connectors: Object.values(connectors),
-      transports: Object.fromEntries(
-        networkEntries.map(([id, config]) => [
-          id,
-          /**
-           * Transport configuration for Wagmi:
-           * 1. unstable_connector(injected) - Uses the user's injected wallet for transactions
-           *
-           *    According to Wagmi docs, it is highly recommended to use unstable_connector
-           *    inside a fallback transport because connector requests may fail due to:
-           *    - Chain ID mismatches
-           *    - Limited RPC method support
-           *    - Rate-limiting of wallet provider
-           *
-           * 2. Multiple http transports - Used as fallback when connector transport fails
-           *    - Primary: http transport with default URL (batch size 3 for DRPC compliance)
-           *    - Fallbacks: Additional http transports for each fallback RPC URL
-           *    - Prevents error code 29 from exceeding batch limits
-           *    - Gives browsers a time window for the opportunity to batch multiple calls
-           *
-           * Note: WalletConnect ignores this transport configuration and uses chain.rpcUrls.default.http
-           * in order, but having multiple transports helps with injected wallet resilience.
-           */
-          fallback([
-            unstable_connector(injected),
-            ...getRpcUrls(id).map((url) => http(url, { batch: { batchSize: 3, wait: 50 } })),
-          ]),
-        ]),
-      ) as Record<ChainId, Transport>,
+  <TChains extends readonly [Chain, ...Chain[]]>({
+    chains,
+    transports,
+    connectors = Object.values(defaultConnectors),
+  }: CreateWagmiConfigOptions<TChains>) =>
+    createConfig({
+      chains,
+      connectors,
+      transports,
       /**
        * Disabled to prevent auto-reconnect on disconnecting an injected wallet.
        * Solves the same issue as discussed on https://github.com/wevm/wagmi/discussions/3537.
        * We don't use wallet specific injected connectors anyway and prefer to use the single global one.
        */
       multiInjectedProviderDiscovery: false,
-    })
-  },
+    }),
   {
     max: 1, // only memoize the last call
+    normalizer: ([{ chains, transports }]) =>
```

Comment:
I think this is only necessary due to the missing useMemo above?

Reply:
Even with the useMemo it's necessary. Took me a lot of debugging to find out I could get rid of all the memos and this normalizer function was literally all that's needed to avoid multiple config instances.